### DISCUSSION TITLE
Getting error for V4.1.0

### DISCUSSION THREAD

**Discussion by Takrel** [cameroncros/OctoPrint-DiscordRemote#241] [General]
Hello

I wanted to update to the newest version. But what i get is this message:

Installiere Plugin von https://github.com/cameroncros/OctoPrint-DiscordRemote/archive/refs/tags/v4.1.0.zip...
/home/pi/oprint/bin/python3 -m pip --disable-pip-version-check install file:///tmp/tmpfdxbwtvj/OctoPrint-DiscordRemote-4.1.0.zip --no-cache-dir
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Processing /tmp/tmpfdxbwtvj/OctoPrint-DiscordRemote-4.1.0.zip
Preparing metadata (setup.py): started
Preparing metadata (setup.py): finished with status 'done'
Requirement already satisfied: pillow in /home/pi/oprint/lib/python3.7/site-packages (from OctoPrint-DiscordRemote==4.1.0) (8.4.0)
ERROR: Ignored the following versions that require a different python version: 2.0.0 Requires-Python >=3.8.0; 2.0.1 Requires-Python >=3.8.0; 2.1.0 Requires-Python >=3.8.0; 2.1.1 Requires-Python >=3.8.0; 2.2.0 Requires-Python >=3.8.0; 2.2.1 Requires-Python >=3.8.0; 2.2.2 Requires-Python >=3.8.0
ERROR: Could not find a version that satisfies the requirement discord.py==2.2.2 (from octoprint-discordremote) (from versions: 0.1.0, 0.2.0, 0.2.1, 0.3.0, 0.3.1, 0.4.0, 0.4.1, 0.5.0, 0.5.1, 0.6.0, 0.6.1, 0.6.2, 0.6.3, 0.7.0, 0.8.0, 0.9.0, 0.9.1, 0.9.2, 0.10.0, 0.11.0, 0.12.0, 0.13.0, 0.14.0, 0.14.1, 0.14.2, 0.14.3, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.16.2, 0.16.3, 0.16.4, 0.16.5, 0.16.6, 0.16.7, 0.16.8, 0.16.9, 0.16.10, 0.16.11, 0.16.12, 1.0.0, 1.0.1, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.2.4, 1.2.5, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.6.0, 1.7.0, 1.7.1, 1.7.2, 1.7.3)
ERROR: No matching distribution found for discord.py==2.2.2
INFO: pip is looking at multiple versions of octoprint-discordremote to determine which version is compatible with other requirements. This could take a while.
Looks like the plugin was already installed. Forcing a reinstall.
/home/pi/oprint/bin/python3 -m pip --disable-pip-version-check install file:///tmp/tmpfdxbwtvj/OctoPrint-DiscordRemote-4.1.0.zip --no-cache-dir --ignore-installed --force-reinstall --no-deps
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Processing /tmp/tmpfdxbwtvj/OctoPrint-DiscordRemote-4.1.0.zip
Preparing metadata (setup.py): started
Preparing metadata (setup.py): finished with status 'done'
INFO: pip is looking at multiple versions of octoprint-discordremote to determine which version is compatible with other requirements. This could take a while.
ERROR: Package 'octoprint-discordremote' requires a different Python: 3.7.3 not in '>=3.8.0'
Fehler!
Unexpected error while trying to install plugin from https://github.com/cameroncros/OctoPrint-DiscordRemote/archive/refs/tags/v4.1.0.zip

I hope you can fix this soon. I really like this plugin.

Kindly regards
Reto

**Comment by cameroncros** [comment]
There is no significant change with the 4.X release, except to support the latest beta versions of octoprint. Upgrading is definitely not required unless you want to 
If you really want to use latest plugin, or you want to use the latest octoprint beta, you will need to upgrade your octoprint virtual environent.

This is non-trivial, and I did manage it on my octopi install, but I definitely wouldnt recommend it.
For me, I had to:

1. System upgrade to the latest debian version.
2. Install latest python
3. Delete /home/pi/oprint (The virtual env, also contains all your installed plugins)
4. Recreate the virtualenv (python3.9 -m venv /home/pi/oprint)
5. Reinstall all the previous plugins.

I dont want to give any more details, as it would probably be better to just backup octoprint and reimage with the latest octopi release.
Of course, if you are using anything slightly non-standard, your mileage may vary.
